lang/cs/Org.Apache.REEF.Network/Group/Task/Impl/OperatorTopology.cs

```diff
@@ -356,6 +393,72 @@ public bool HasChildren()
         }
 
         /// <summary>
+        /// Get a set of node containing an incoming message and belonging to candidate set of nodes.
+        /// </summary>
+        ///<param name="nodeSet">Candidate set of nodes</param>
+        /// <returns>A NodeStruct with incoming data.</returns>
+        /// private IEnumerable<NodeStruct> GetNodeWithData(IEnumerable<string> nodeSet)
+        private IEnumerable<NodeStruct> GetNodeWithData(IEnumerable<string> nodeSetIdentifier)
+        {
+            CancellationTokenSource timeoutSource = new CancellationTokenSource(_timeout);
+            List<NodeStruct> nodesSubsetWithData = new List<NodeStruct>();
+     
+            try
+            {
+                lock (_thisLock)
+                {
+                    foreach (var identifier in nodeSetIdentifier)
+                    {
+                        if (_idToNodeMap[identifier].HasMessage())
+                        {
+                            nodesSubsetWithData.Add(_idToNodeMap[identifier]);
+                        }
+                    }
+
+                    if (nodesSubsetWithData.Count > 0)
+                    {
+                        return nodesSubsetWithData;
```

Comment:
yes....it just give nodeIDs..actual taking of data takes place in ReceiveFromChildren